### ISSUE TITLE
getROSbin.py not support npk 5 version?

### ISSUE THREAD

**Issue by skysider** [BigNerd95/Chimay-Red#38]
R.T.
I try to use the function get_binary in the script to extract npk file from http://files.shelbybb.com/mikrotik/5.19/ and it doesn't work.

**Comment by BigNerd95** [comment]
Since chimay red doesn't support 5.x I made getROSbin for 6.x only

**Comment by BigNerd95** [comment]
Then I made getROSbin to speed up the process only.
If you want to audit other versions you can use binwalk

**Comment by skysider** [comment]
Indeed, I test two 5.x version firmware with poc of dos mode, it works.

**Comment by BigNerd95** [comment]
If something crashes doesnt mean it is exploitable
I didnt tested 5.x but another researcher told me that www on 5.x is not multithreaded so the exploit must rewritten

**Comment by Hume3** [comment]
@skysider 
> I try to use the function get_binary in the script to extract npk file from http://files.shelbybb.com/mikrotik/5.19/ and it doesn't work.

get_binary() use Squashfs 4.0 decompressor.
5.x .npk files use Zlib compression

**binwalk 5x.npk**:  Zlib compressed data, default compression

You can see how decompression is done in this part, 
then you just have to extract the path /nova/bin/www
[https://github.com/kost/mikrotik-npk/blob/d54e97caac9ea447e29939ca4176d17eeff856a9/unpacknpk.py#L90](https://github.com/kost/mikrotik-npk/blob/d54e97caac9ea447e29939ca4176d17eeff856a9/unpacknpk.py#L90)

**Comment by skysider** [comment]
@Hume3 thanks, I will have a look.

**Comment by dd404x** [comment]
@Hume3 thanks,I used your method to successfully unzip the npk file。But after this I don't know how to operate？
![image](https://user-images.githubusercontent.com/19923974/38233634-cd42dd94-374d-11e8-9dae-1cd87764b969.png)
![image](https://user-images.githubusercontent.com/19923974/38233652-d88be510-374d-11e8-8137-d7ff0554b478.png)
![image](https://user-images.githubusercontent.com/19923974/38233701-081ef9c0-374e-11e8-9d0c-fedb4f27d696.png)
